classes/i18n/PKPLocale.inc.php

```diff
@@ -147,30 +147,50 @@ static function initialize($request) {
 
 		if (Config::getVar('general', 'installed')) {
 			// Set the time zone for DB
-			// Get the offset from UTC
-			$now = new DateTime();
-			$mins = $now->getOffset() / 60;
-			$sgn = ($mins < 0 ? -1 : 1);
-			$mins = abs($mins);
-			$hrs = floor($mins / 60);
-			$mins -= $hrs * 60;
-			$offset = sprintf('%+d:%02d', $hrs*$sgn, $mins);
-
 			$conn = DBConnection::getInstance();
 			$dbconn =& $conn->getDBConn();
-			switch($conn->getDriver()) {
+
+			// It looks like ADO can't take any composed driver. i.e. pdo_mysql, pdo_mssql...
+			$driver = $conn->getDriver();
+			if (strpos($driver, 'pdo') !== -1) {
+				$driver = 'pdo';
+			}
+
+			switch($driver) {
 				case 'mysql':
 				case 'mysqli':
+					$offset = PKPLocale::getDateOffset();
```

Comment:
@mfelczak, since you tackled https://github.com/pkp/pkp-lib/issues/1031, would you mind taking a look at this?